tests/temperature_monitor_ut.c

```diff
@@ -0,0 +1,219 @@
+#include "pal.h"
+#include "monitor.h"
+#include <stdio.h>
+#include <stdbool.h>
+
+/**
+ * Session of mock functions that will replace the PAL ones.
+ */
+static pal_led_color g_led_color;
+static pal_result g_pal_led_set_color_result;
+pal_result pal_led_set_color(pal_led_color color)
+{
+    printf("This is the mock pal_led_set_color\r\n");
+    g_led_color = color;
+    return g_pal_led_set_color_result;
+}
+
+static unsigned short g_temperature;
+static pal_result g_pal_temperature_get_result;
+pal_result pal_temperature_get(unsigned short* temperature)
+{
+    printf("This is the mock pal_temperature_get\r\n");
+    *temperature = g_temperature;
+    return g_pal_temperature_get_result;
+}
+
+float pal_temperature_to_celsius(unsigned short temperature) {
+    return ((float)(temperature - 80) / 2.0f);
+}
+
+static int g_speed_rpm;
+static pal_result g_pal_fan_set_speed_result;
+pal_result pal_fan_set_speed(int speed_rpm) {
+    printf("This is the mock pal_fan_set_speed\r\n");
+    g_speed_rpm = speed_rpm;
+    return g_pal_fan_set_speed_result;
+}
+
+static bool g_pal_halt;
+void pal_halt(void) {
+    printf("This is the mock pal_halt... HALT!\r\n");
+    g_pal_halt = true;
+}
+
+void pre_test(void) {
+    g_led_color = PAL_LED_OFF;
+    g_pal_led_set_color_result = PAL_OK;
+    g_temperature = 80;
+    g_pal_temperature_get_result = PAL_OK;
+    g_speed_rpm = 0;
+    g_pal_fan_set_speed_result = PAL_OK;
+    g_pal_halt = false;
+}
+
+void post_test(void) {
+
+}
+
+/**
+ * Helper macro to execute tests.
+ */
+#define RUM_TEST(test, count_error) \
```

Comment:
```suggestion
#define RUN_TEST(test, count_error) \
```

Reply:
?

---
In reply to: [543626192](https://github.com/mamokarz/sample-multi-platform/pull/2#discussion_r543626192) [](ancestors = 543626192)